op-challenger/game/scheduler/scheduler.go

```diff
@@ -48,13 +55,28 @@ func NewScheduler(logger log.Logger, m SchedulerMetricer, disk DiskManager, maxC
 	}
 }
 
+func (s *Scheduler) ThreadActive() {
+	s.executorMutex.Lock()
+	defer s.executorMutex.Unlock()
+	s.m.DecIdleExecutors()
+	s.m.IncActiveExecutors()
```

Comment:
You shouldn't need a mutex here - metrics are thread-safe for updates, and the lock won't stop the values being sampled between these two calls anyway.  Just do the increment before the decrement in both cases so its impossible to wind up with a negative value in the gauge - although I think prometheus is ok with that anyway.